### DISCUSSION TITLE
Painting - what can be painted?

### DISCUSSION THREAD

**Discussion by John-Nagle** [emilk/egui#4266] [Q&A]
Painter has functions to fill rectangles and circles, and to paint text. That's all. It seems to use epaint, though, which has PathShape and similar more general paints. Is it possible to use those with Painter? If so, how? Thanks.

**Comment by emilk** [comment]
You can use [`Painter::add`](https://docs.rs/egui/latest/egui/struct.Painter.html#method.add) to add any `Shape`
